Dockerfile

```diff
@@ -29,25 +28,23 @@ ARG DEBIAN_FRONTEND="noninteractive"
 ARG DEBCONF_NOWARNINGS="yes"
 
 # install system dependencies
-ARG SYS_DEPS="\
-    git \
-    libcurl4-openssl-dev \
-    libssl-dev \
-    openssh-client \
-    wget \
-    "
 RUN apt-get update \
-    && apt-get install -y --no-install-recommends $SYS_DEPS \
+    && apt-get install -y --no-install-recommends \
+      git=1:2.34.* \
+      libcurl4-openssl-dev=7.81.* \
+      libicu-dev=70.* \
```

Comment:
I would strongly suggest allowing pak to install system dependencies that are needed by R packages. I've run into problems with getting R pkgs to properly locate system libraries when they're installed in some system defined location and not where the pkg is expecting them. Installing with pak avoids that problem.

Reply:
I think this might be a philosophical difference, but I'm generally of the mindset that `pak` should inform us when a dependency is missing, but not actually touch the system, (only the `R` library space).

It's important to have a clear understanding of what system packages are being installed (see above where `pak` is installing `pandoc` unexpectedly).

Reply:
> I think this might be a philosophical difference, but I'm generally of the mindset that `pak` should inform us when a dependency is missing, but not actually touch the system, (only the `R` library space).
> 
> It's important to have a clear understanding of what system packages are being installed (see above where `pak` is installing `pandoc` unexpectedly).

hehe, yeah.... I would consider that a feature, not a problem. I've had so many struggles with getting R pkgs linked properly to system libraries that I prefer to let people that have done a bunch of testing and investigation into specifically this problem handle it rather than letting some linux flavor authors choose an arbitrary location for the needed libraries.

Reply:
when we get to installing LaTeX stuff, I'm even more adamant about that, because it can easily become a major headache

Reply:
Also, `pak` doesn't do any of the `apt` cleanup when its done. Not a huge issue, but just another thing in the whole "let the system package manager handle system packages, and the R package manager handle R packages" philosophy

Reply:
technically, pak uses the system package manager, so there's not really any difference... it just knows better than we do exactly *how* to use the system package manager so it does things correctly for it to be available to the R pkgs

Reply:
I don't think this is a hill worth dying on one way or the other, but I've never experienced issues with library paths when `apt` handles the installation, and this is a good way to ensure we have stable versions of the sysdeps, rather than pak -> apt -> grabbing whatever is available.

Reply:
@cjyetman I opened #17 which calls `pak::sysreqs_check_installed()`. I think that's a reasonable tradeoff here: we're using `pak` to recognize if the sysdeps are installed or not (which mirrors the behavior we would get if `pak` were doing the installations), but we can still manage the dependencies explicitly (and leverage build cache using the changes from #12 ).

Reply:
If I approve this now, will you promise to not force us to do this when it comes to adding LaTeX stuff? I'm thinking about the long term... if we commit to this now, we're setting ourselves up for trouble in the future.

Reply:
🤔 Yeah LaTeX is always a pain. espescially with `tlmgr` paths. I promise to keep an open mind if this setup isn't amenable to working with LaTeX, but I'm hoping that we can keep consistency in how we approach sysdeps

FWIW, I'm aiming to do pacta-report next, so I'll find out soon if it works or not. `:lolsob:`

Reply:
Please also recognize that committing to this also forces us to... whenever a new R pkg dependency is introduced we need to (correctly) identify the system dependencies and remember to manually add them here (the new dependency will show up in a different repo, and then this repo will be broken until it's adequately updated). When I switched to installing with pak some months ago, one of my primary reasons for doing so was to avoid exactly that, after many discussion with Jackson about managing our dependencies. Now we're effectively reverting all of that effort. I'm really annoyed by having all these interdependent repos that break if we change one thing somewhere else, and then we need to orchestrate this multi step update process for multiple repos in a very specific order.

Reply:
Yeah, I hear that concern about managing dependencies across repos. But to me that friction is reasonable, since it forces us to be explicit about adding dependencies (or in this case, at least dependencies that require new sysdeps). I think the clarity of being explicit in our dependencies outweighs the cost of needing to check the dockerfile when a new dep gets added.

Reply:
and if we find that it really is a problem, we can switch back to the default `pak` behavior.